code/datums/components/storage/storage.dm

```diff
@@ -829,10 +828,14 @@
 	return hide_from(target)
 
 /datum/component/storage/proc/on_right_click(datum/source, mob/user)
-	SIGNAL_HANDLER
+	SIGNAL_HANDLER_DOES_SLEEP
```

Comment:
No new sleeping signal handlers are allowed. Yes, I know you're transferring this over, but no better time than now--move the sleeping part (probably put_in_hands) and everything after in its own proc and invoke that asyncly with INVOKE_ASYNC.